src/Core/Services/ResolverMiddleware.cs

```diff
@@ -62,6 +66,12 @@ public async Task InvokeAsync(IMiddlewareContext context)
 
             if (context.Selection.Field.Coordinate.TypeName.Value == "Mutation")
             {
+                // Track event when request is received
+                if (_telemetryClient is not null)
+                {
+                    TrackEventForGraphQLRequests(context, _telemetryClient);
+                }
```

Comment:
Should these events come in a separate PR? I'd imagine we'd want to allow users to customize which events are logged, if they want any additional events at all.

For a highly utilized app, this event (and others) would flood the logs and increase costs to use AppInsights.

Reply:
makes sense, will move this to a seperate PR.

Reply:
@seantleonard how do we, in future see allowing user to customize events that are logged? Via config file?

Reply:
Yeah, in cli we will add a command to add telemetry, there we can give them a switch to enable some of the predefined event tracking feature embedded in DAB, for more customization. They can create an issue or create their own custom events inside the code.

Reply:
I see, but providing ability to them to create custom event means adding logging to different places in the code, right? I know that's looking too far ahead, but won't that be a bit tricky? Are such abilities provided by any other APIs that you know of?

Reply:
Not that I'm aware of, trackEvent is mostly to capture different features within the code or to monitor performance of specific section of code, such as a graphQL request or database operation, etc. For this PR, i won't be keeping this custom trackEvents and will check if there are better ways to do it.

Reply:
I see Ayush's point. My perspective is that we should be careful with what logging level we assign to various logging events. If we add a lot of error events to a hot code path, then that greatly increases the number of logs customers have to store. 
In other cloud services, I have seen Microsoft recommend that customers do not have "Trace/verbose/informational" logs in production, but instead experiment with enabling critical errors.

Reply:
Customers are sensitive (as they should be) to increased logging added unnecessarily.. leading to rising costson their part without notice/documentation/guidance from microsoft.

Reply:
> I see Ayush's point. My perspective is that we should be careful with what logging level we assign to various logging events. If we add a lot of error events to a hot code path, then that greatly increases the number of logs customers have to store. In other cloud services, I have seen Microsoft recommend that customers do not have "Trace/verbose/informational" logs in production, but instead experiment with enabling critical errors.

Currently, DAB follows the same approach, in Production environment, we don't generate console logs below error level and same for application insights we don't send logs below error level.

Also, A cutom event for tracking errors will only be sent for errors.